### PR TITLE
Maintenance: Update CFG_CAL_DAC to Charge Conversion for v3

### PR DESCRIPTION
**Pull request by bdorney**
## Description

The `vfat3` team has decided to change the convention for the calibration coefficients `calDAC2Q_m` and `calDAC2Q_b`.  Previously in voltage step mode the "inverted" `CFG_CAL_DAC` register value was used, i.e.:

```
Q = calDAC2Q_m * (256 - CFG_CAL_DAC) + calDAC2Q_b
``` 

Now the raw register value will be used, e.g.:

```
Q = calDAC2Q_m * (CFG_CAL_DAC) + calDAC2Q_b
```

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Maintenance (updates to the software that are transparent to the user)

## Motivation and Context

This was reported by Henri who is developing the `vfat3` production testing:

>My idea was to make the fit on the raw register value so the charge equation for the voltage mode would become:
>Q=m*CAL_DAC+b
>which in my mind would be more straight forward and avoid confusion about when the inversion is used and when not.

## How Has This Been Tested?

Took a new scurve and analyzed the data.

### Screenshots (if appropriate):
summary

Note VFAT5 & 12 were masked in this test.

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: fitting/fitScanData.py [hunk 1]
```diff
@@ -68,7 +74,7 @@ def feed(self, event):
                 if(event.vcal > 254):
                     self.scanCount[event.vfatN][event.vfatCH] += event.Nhits
             else:
-                charge = self.calDAC2Q_m[event.vfatN]*(256-event.vcal)+self.calDAC2Q_b[event.vfatN]
+                charge = self.calDAC2Q_m[event.vfatN]*(event.vcal)+self.calDAC2Q_b[event.vfatN]
                 if((256-event.vcal) > 254):
```
Comment (BenjaminRS) [inline fitting/fitScanData.py]: Hi Brian, I am (very slowly) still looking into bug #61 and so have been converting charge to chargebin later in this script. Here should the 256 not also be removed from this line? So it becomes identical to L74?

Comment (bdorney) [reply fitting/fitScanData.py]: In this particular case no.  We are trying to count the number of `scurves` that have been processed.  So all we are interested in is the raw register value.  In the case of voltage step mode (e.g. `isCurrentPulse == False`) the height of the cal pulse is high when the register value is low.

File: fitting/fitScanData.py [hunk 2]
```diff
@@ -119,8 +125,12 @@ def fit(self, debug=False):
         random = r.TRandom3()
         random.SetSeed(0)
         for vfat in range(0,24):
-            fitTF1 = r.TF1('myERF','[3]*TMath::Erf((TMath::Max([2],x)-[0])/(TMath::Sqrt(2)*[1]))+[3]',
-                        self.calDAC2Q_m[vfat]*1+self.calDAC2Q_b[vfat],self.calDAC2Q_m[vfat]*253+self.calDAC2Q_b[vfat])
+            if self.isVFAT3:
+                fitTF1 = r.TF1('myERF','[3]*TMath::Erf((TMath::Max([2],x)-[0])/(TMath::Sqrt(2)*[1]))+[3]',
+                            self.calDAC2Q_m[vfat]*253+self.calDAC2Q_b[vfat],self.calDAC2Q_m[vfat]*1+self.calDAC2Q_b[vfat])
```
Comment (mexanick) [inline fitting/fitScanData.py]: Please forgive me my ignorance, but how the factor `253` appeared here?
and also why do we need multiplication by factor `1`: `self.calDAC2Q_m[vfat]*1`?

Comment (bdorney) [reply fitting/fitScanData.py]: We need the axis to be in charge units.  With the updated convention Henri has put forward `calDAC2Q_m` will be negative and work with the raw `CFG_CAL_DAC` register.  In voltage pulse mode low `CFG_CAL_DAC` is high charge, and high `CFG_CAL_DAC` is low charge.  The factors [253,1] then specify the charge daq range which is something like [0, 60] fC.

**Pull request merged**